### PR TITLE
 CRM-21677 reduce unnecessary joins in Case, Grant, Survey reports

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Reviewers cut of #11550 - reduce unnecessary joins to address tables in reports
Campaign/SurveyDetails
Case/Demographics
Case/Detail
Grant/Detail
Grant/Statistics

Before
----------------------------------------
unnecessary joins to phone, email and address tables

After
----------------------------------------
join only when needed, code standardised

Technical Details
----------------------------------------
This also aligns the approach with that used in the extended report extension - there might be some notices as I have updated the extension where the signature differs from core but will need to drop a new release & people will need to be on it.

Comments
----------------------------------------
@yashodha back to you :-) This takes us down to 15 reports

---

 * [CRM-21677: Report improvements](https://issues.civicrm.org/jira/browse/CRM-21677)

### CODE REVIEW

**Comment by jitendrapurohit** [comment]
reviewing this now

File: CRM/Report/Form/Grant/Statistics.php [hunk 1]
```diff
@@ -241,18 +232,14 @@ public function from() {
...
+    if ($this->isTableSelected('civicrm_worldregion')) {
       $this->_from .= "
-                  LEFT JOIN civicrm_address {$this->_aliases['civicrm_address']}
-                         ON {$this->_aliases['civicrm_contact']}.id =
-                            {$this->_aliases['civicrm_address']}.contact_id AND
-                            {$this->_aliases['civicrm_address']}.is_primary = 1\n
-                  LEFT JOIN civicrm_country country
-                         ON {$this->_aliases['civicrm_address']}.country_id =
-                            country.id
-                  LEFT JOIN civicrm_worldregion {$this->_aliases['civicrm_world_region']}
```
Comment (jitendrapurohit) [inline CRM/Report/Form/Grant/Statistics.php]: This results into DB error on Grant Statistics report. 
> Database Error Code: Unknown column 'country.region_id' in 'on clause', 1054

Maybe, we should keep the left join of `civicrm_country` to connect with worldregion table?

Comment (eileenmcnaughton) [reply CRM/Report/Form/Grant/Statistics.php]: thanks - I'll take a look....

Comment (eileenmcnaughton) [reply CRM/Report/Form/Grant/Statistics.php]: @jitendrapurohit I pushed an update - altered start of country join to

```
  protected function joinCountryFromAddress($prefix = '', $extra = array()) {
    // include country field if country column is to be included
    if ($this->isTableSelected($prefix . 'civicrm_country') || $this->isTableSelected($prefix . 'civicrm_worldregion')) {
      if (empty($this->_aliases[$prefix . 'civicrm_country'])) {
        $this->_aliases[$prefix . 'civicrm_country'] = $prefix . '_report_country';
      }
```

**Review by jitendrapurohit — APPROVED** [review]
Noted issue has been fixed. Tested all the reports. They seems to work fine.

**Pull request merged**